File: Task 1/C++/Components/Product/Plants.cpp
#include "Plants.h"
Plants::Plants() : Flora(){
    bearsFruit = false;
}
Plants::Plants(std::string name, float cost, std::string supplier, float discount_amount, int packing_units, float cost_per_unit, bool isCultivated, bool bearsFruit) 
        : Flora(name, cost, supplier, discount_amount, packing_units, cost_per_unit, isCultivated){
    this->bearsFruit = bearsFruit;
}

Plants::Plants(const Plants& other) : Flora(other) {
    this->bearsFruit = other.bearsFruit;
}

std::string Plants::display_product_info(){
    return Flora::display_product_info()+
            "\nBears Fruit: " +(bearsFruit? "true" : "false");
}

File: Task 1/Java/logistics/src/main/java/com/logistics/Components/Product/Plants.java
package com.logistics.Components.Product;

public class Plants extends Flora {

    private boolean bearsFruit;

    public Plants() {
        super();
        bearsFruit = false;
    }

    public Plants(String name, float cost, String supplier, float discountAmount, int packingUnits, float costPerUnit, boolean isCultivated, boolean bearsFruit) {
        super(name, cost, supplier, discountAmount, packingUnits, costPerUnit, isCultivated);
        this.bearsFruit = bearsFruit;
    }

    // Copy Constructor
    public Plants(Plants other) {
        super(other);
        this.bearsFruit = other.bearsFruit;
    }

    // Clone method
    @Override
    public Plants clone() {
        return new Plants(this);
    }

    // Getters
    public boolean getBearsFruit() { return bearsFruit; }

    // Setters
    public void setBearsFruit(boolean bearsFruit) { this.bearsFruit = bearsFruit; }

    @Override
    public String display_product_info() {
        return super.display_product_info() + 
                "\nBears Fruit: " + bearsFruit;
    }
}
